packages/sdk/odahuflow/sdk/local/training.py

```diff
@@ -101,22 +124,27 @@ def start_train(trainer: K8sTrainer, output_dir: str):
     create_mt_config_file(trainer)
 
     if not output_dir:
-        # For example, 01-Mar-2020-17-38-04
-        suffix = datetime.datetime.now().strftime("%d-%b-%Y-%H-%M-%S")
-        output_dir = join(
-            config.LOCAL_MODEL_OUTPUT_DIR,
-            f'{trainer.model_training.spec.model.name}-{trainer.model_training.spec.model.version}-{suffix}'
-        )
+        output_dir = config.LOCAL_MODEL_OUTPUT_DIR
+        LOGGER.debug(f'Output directory for model training is not provided. Using default: {output_dir}')
 
-        LOGGER.debug(f'Output model training directory is not provided. Generate the directory: {output_dir}')
+    model_dir_name_template = trainer.model_training.spec.model.artifact_name_template or DEFAULT_MODEL_DIR_TEMPLATE
 
-    os.makedirs(output_dir, exist_ok=True)
+    model_dir_name_template = model_dir_name_template.rstrip('.zip')
+
+    model_dir_name = compile_artifact_name_template(
+        go_template=model_dir_name_template,
+        context=TemplateContext(Name=trainer.model_training.spec.model.name,
+                                Version=trainer.model_training.spec.model.version,
+                                RandomUUID=uuid.uuid4())
+    )
 
-    launch_training_container(trainer, output_dir)
+    model_dir_path = os.path.join(output_dir, model_dir_name)
+    os.makedirs(model_dir_path, exist_ok=True)
```

Comment:
it also, maybe, worth to add a check on exceptions OSError, PerimissionError. In case of bad permissions or invalid dir name.

Reply:
agree

Reply:
OK, agree